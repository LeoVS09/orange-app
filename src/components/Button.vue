<template>
    <div class="button-container">
      <button class="button-container--submit"><slot>Submit</slot></button>
    </div>
</template>

<script lang="ts">
  import Vue from 'vue'
  import {Component} from 'vue-property-decorator'

  interface Options {
    tabindex?: number
  }

  @Component({
    props: {
      tabindex: Number
    }
  })
  export default class Button extends Vue {
    get options(): Options {
      let result: Options = {};
      // @ts-ignore
      if(this.tabindex) {
        // @ts-ignore
        result['tabindex'] = this.tabindex;
      }
      return result;
    }
  }
</script>

<style scoped lang="scss">
  @import "../styles/config.scss";

  .button-container {
    &--submit {
      margin-top: 6rem;
      padding: 0.5rem 5rem;
      border-radius: 1rem;
      background-color: $buttonColor;
      border: none;
      box-shadow: 0 0 3px 0 lighten($buttonColor, 10%);
      color: white;
      cursor: pointer;
      transition-property: box-shadow;
      transition-duration: 0.1s;

      &:hover {

        background-color: lighten($buttonColor, 3%);
        box-shadow: 0 0 10px 0 lighten($buttonColor, 10%);
      }

      &:active {
        background-color: $activeColor;
      }
    }
  }
</style>
