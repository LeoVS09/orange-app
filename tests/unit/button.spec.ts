import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Button.vue'

describe('Button.vue', () => {
  it('renders slot when passed', () => {
    const msg = 'Click on me'
    const wrapper = shallowMount(Button, {
      slots: { default: msg }
    })
    expect(wrapper.text()).toMatch(msg)
    expect(wrapper.element).toMatchSnapshot()
  })
})
