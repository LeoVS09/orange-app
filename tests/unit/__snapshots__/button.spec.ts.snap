// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders slot when passed 1`] = `
<div
  class="button"
>
  <button
    class="button--submit gradient-highlight text-not-faded"
  >
    <div
      class="button--content"
    >
      <!---->
       
      <transition-stub
        name="text-fade"
      >
        <span
          class="button--text"
        >
          Click on me
        </span>
      </transition-stub>
       
      <!---->
    </div>
  </button>
</div>
`;
